### PR TITLE
Fix sliders

### LINKED ISSUE #312: behavior of long press for lights?
What is the intended behavior for long-press of (at least) lights? I experience that on long-press my light switches on. And a bit unreliably, the light slider is shown. I'd expect the light slider to show without the light switching on.

### PR DESCRIPTION
**Pull request by akloeckner**
* make them appear also if entity is off (hacky)
* fix their color to match theme

Hacky code is for you to test. I'll simply delete the passages if approved.

fixes #312

My first idea for the colors was to add the item class to the sliders. Would that be better?

### CODE REVIEW

**Comment by resoai** [comment]
If the light is off, than no attributes are available. This is why the logic was to turn the light on and than, after a short delay, to show the controls.

**Comment by akloeckner** [comment]
> If the light is off, than no attributes are available. 

As said in #312, that's the case for me too. But the `turn_on` service accepts a brightness also if the light is off. So, I figure it reasonable to allow switching on directly to a dimmed state. It works perfectly for me. The PR is meant for you to test, if it also works for you. Does it? 

Btw. If, in the old logic, I open up the slider, then dim to zero, the light turns off but the slider stays and I can turn it on again. If we force that old (opinionated, you'd say ;-) ) logic on the user, it should be done consistently. Meaning: the slider should disappear, if the light is turned off.

**Comment by akloeckner** [comment]
I just checked: the main code already caters for the case of no attributes being available in the most sensible way, I can think of, with fallback and default:

https://github.com/resoai/TileBoard/blob/1a09a29a2c0b46d551424e645ede12cb46aa5a4f/scripts/controllers/main.js#L778

So, switching on the light first seems like an overkill, handling the same case twice.

**Comment by resoai** [comment]
When we are switching light on, we don’t know what is the percentage of brightness. We should always keep it at the level it was when light was turned off.

**Comment by rchl** [comment]
The part of the change that fixes sliders tile styling is great but shouldn't be mixed with the hack that obviously can't be integrated as-is.

**Comment by rchl** [comment]
> When we are switching light on, we don’t know what is the percentage of brightness. We should always keep it at the level it was when light was turned off.

I can see the use case for turning on the light with given brightness level though. For example at night when you don't want to blind everyone initially. :)

**Comment by resoai** [comment]
> > When we are switching light on, we don’t know what is the percentage of brightness. We should always keep it at the level it was when light was turned off.
> 
> I can see the use case for turning on the light with given brightness level though. For example at night when you don't want to blind everyone initially. :)

While being true, this is not the expected behaviour and not how it works in HA.

**Comment by resoai** [comment]
I would honestly just start from scratch and create a popup on a long press with sliders and colour wheel. Current implementation is plain bad and nightmare for new users to configure.

**Comment by akloeckner** [comment]
> We should always keep it at the level it was when light was turned off.

True. That would be done by clicking the tile. If you want to change brightness, you can then long-press and select. Or long-press and select brightness directly.

> So not needed on master by itself.

I don't get that. But yes, you're right: I also thought the old behavior was a bug. Until I found out, it was explicitly programmed that way. I'm still convinced, the new logic is the better one.

**Comment by akloeckner** [comment]
> I would honestly just start from scratch 

Also fine. Just please have it be possible to (as @rchl says) to not blind everyone at night... :-)

File: styles/main.less [hunk 1]
```diff
@@ -65,7 +65,7 @@
 @DIMMER_SWITCH: rgb(130, 156, 108);
 
 
-.item {
+.item, .item-entity-sliders {
```
[the remark this answers is not in the document]
Comment (akloeckner) [reply styles/main.less]: I believe, it would work. I just didn't want to mix things up.

**Comment by rchl** [comment]
Seems like HA is actually showing brightness slider for light that is off (only color temperature disappears). I guess I've checked wrongly before.
Screenshot 2019-12-04 at 22 19 02

**Comment by resoai** [comment]
Indeed, it only exposes brightness being set to zero (and not colour temperature etc) but only for lights that support this feature. We have a function to do that:

1. https://github.com/resoai/TileBoard/blob/master/scripts/controllers/main.js#L877
2. https://github.com/resoai/TileBoard/blob/master/scripts/app.js#L77

I suppose it does make sence allowing light to turn on by setting brightness if feature is supported.

**Comment by akloeckner** [comment]
I'm not sure, I get what you mean. Do you suggest to check for `supportsFeature(brightness)` in this line? 

https://github.com/akloeckner/TileBoard/blob/1a09a29a2c0b46d551424e645ede12cb46aa5a4f/scripts/controllers/main.js#L832

Currently, the slider is only there, if the user explicitly defines it. Can't it be safe to assume, the light supports a brightness then? 

Or do you suggest to hide any slider (except the brightness, if supported) if the light is off? This seems like an overly complicated logic for something that the user has already defined explicitly via an option.

**Comment by resoai** [comment]
I thought that we are starting from scratch here. Defining sliders manually was not a good idea.

File: styles/main.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (akloeckner) [reply styles/main.less]: I tried. It works.

**Comment by akloeckner** [comment]
Oh, ok. I'll hold back then...

**Comment by akloeckner** [comment]
closing this in favor of #388 or something similar

**Pull request closed without merging**